cmd/zpool/zpool_main.c

```diff
@@ -7253,15 +7253,59 @@ print_error_log(zpool_handle_t *zhp)
 	elem = NULL;
 	while ((elem = nvlist_next_nvpair(nverrlist, elem)) != NULL) {
 		nvlist_t *nv;
-		uint64_t dsobj, obj;
+		uint64_t dsobj, obj, data_block_size, indirect_block_size;
+		uint64_t *block_ids;
+		int64_t *indrt_levels;
+		unsigned int error_count;
 
 		verify(nvpair_value_nvlist(elem, &nv) == 0);
 		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_DATASET,
 		    &dsobj) == 0);
 		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_OBJECT,
 		    &obj) == 0);
-		zpool_obj_to_path(zhp, dsobj, obj, pathname, len);
-		(void) printf("%7s %s\n", "", pathname);
+		verify(nvlist_lookup_int64_array(nv, ZPOOL_ERR_LEVEL,
+		    &indrt_levels, &error_count) == 0);
+		verify(nvlist_lookup_uint64_array(nv, ZPOOL_ERR_BLOCKID,
+		    &block_ids, &error_count) == 0);
+
+		zpool_obj_to_path(zhp, dsobj, obj, pathname, len,
+		    &data_block_size, &indirect_block_size);
```

Comment:
I think it would be cleaner to have a separate function that returns these values, rather than tacking them onto zpool_obj_to_path (which, by its name, has a pretty clearly defined role).